### PR TITLE
[8.x] Fix postgresql boolean bind value

### PR DESCRIPTION
**Pull request by jstacho**
Doing a query in postgresql connection the value was bounded as PDO::PARAM_INT
After a php bugfix https://bugs.php.net/bug.php?id=80892 doing a query with boolean value results in a query exception:

"Undefined function: 7 ERROR: operator does not exist: boolean = integer"

### CODE REVIEW

**Comment by taylorotwell** [comment]
Why did you add the `prepareBindings` method regarding dates to solve an issue with booleans?

**Comment by taylorotwell** [comment]
Is this going to break applications that do not have that PHP bug fix yet?

**Comment by jstacho** [comment]
> Why did you add the `prepareBindings` method regarding dates to solve an issue with booleans?

because there is an explicit cast to int for boolean in the parent class: https://github.com/laravel/framework/blob/8.x/src/Illuminate/Database/Connection.php#L611

**Comment by jstacho** [comment]
> Is this going to break applications that do not have that PHP bug fix yet?



> Is this going to break applications that do not have that PHP bug fix yet?

nope, but applications may break when the recent php update ist installed

**Comment by taylorotwell** [comment]
This bug fix will be released with PHP 8.0.4?

**Comment by jstacho** [comment]
> This bug fix will be released with PHP 8.0.4?

has already been released as 8.0.5 https://github.com/php/php-src/releases/tag/php-8.0.5

**Comment by taylorotwell** [comment]
This will indeed break existing applications that query `smallint` columns that are being used as "flags" via booleans:

image

image

image

**Comment by GrahamCampbell** [comment]
This bug will never be fixed on PHP 7.3, though, and Laravel needs to support that?

**Comment by mbeccati** [comment]
@jstacho I'm trying to see if I can somehow keep the bug fix while maintaining some backwards compatibility. Is there some automated test suite I can run to check the behaviour here?

**Review by mfn — COMMENTED** [review]
I use Postgres and can offer to test this, making sure it doesn't break stuff.

File: src/Illuminate/Database/PostgresConnection.php [hunk 1]
```diff
@@ -41,6 +44,28 @@ public function bindValues($statement, $bindings)
         }
     }
 
+    /**
+     * Prepare the query bindings for execution.
+     *
+     * @param  array  $bindings
+     * @return array
+     */
+    public function prepareBindings(array $bindings)
```
[the remark this answers is not in the document]
Comment (jstacho) [reply src/Illuminate/Database/PostgresConnection.php:53]: the idea was to remove the `int` cast

https://github.com/laravel/framework/blob/8.x/src/Illuminate/Database/Connection.php#L611

image

[the remark this answers is not in the document]
Comment (jstacho) [reply src/Illuminate/Database/PostgresConnection.php:53]: do you think it should rather be implemented in the Grammar?

**Comment by mbeccati** [comment]
FYI I've just reverted the ext/pdo_pgsql change on all php-src branches. I really wish this would have popped up earlier during the RC phase, but I have alerted the release managers anyway.

**Comment by taylorotwell** [comment]
@mbeccati thanks for your help on this

**Pull request closed without merging**